File: sentimental-cash/cash.py
from cs50 import get_float

while True:
    # prompt user for amount of change
    change = get_float("Change owed: ")

    # accept only positive int no greater than 8
    if change > 0:
        break

change = 100 * change

# use all 25 cents possible
coin_used_25 = change // 25
remain = change - 25 * coin_used_25

# use all 10 cents possible
coin_used_10 = remain // 10
remain = remain - 10 * coin_used_10

# use all 5 cents possible
coin_used_5 = remain // 5
remain = remain - 5 * coin_used_5

# remaining are all 1 cents
coin_used = coin_used_25 + coin_used_10 + coin_used_5 + remain

print(int(coin_used))

File: cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)  
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    
    int cents = round(dollars * 100);
    int change = 0;
    while (cents >= 25)
    {
        cents = cents - 25;
        change = change + 1;
    }
    
    while (cents >= 10)
    {
        cents = cents - 10;
        change = change + 1;
    }
        
    while (cents >= 5)
    {
        cents = cents - 5;
        change = change + 1;
    }
            
    while (cents >= 1)
    {
        cents = cents - 1;
        change = change + 1;
    }
 
    printf("%i\n", change);
}
